build.sbt

```diff
@@ -103,27 +102,6 @@ def baseSettings: Seq[Setting[?]] = Seq(
   ideSkipProject := scalaVersion.value != defaultScalaVersion,
 )
 
-def compilerVersionDependentScalacOptions: Seq[Setting[?]] = Seq(
-  scalacOptions := {
-    scalaBinaryVersion.value match {
-      case "2.12" | "2.13" =>
-        scalacOptions.value ++ List(
-          "-opt-inline-from:<sources>",
-          "-opt:l:inline",
-          "-Yopt-inline-heuristics:at-inline-annotated"
-        )
-      case _ =>
-        scalacOptions.value.filterNot(
-          Set(
-            "-Xfatal-warnings",
-            "-deprecation",
-            "-YdisableFlatCpCaching",
-          )
-        )
```

Comment:
Don't we have to add this back?

Reply:
For `-Xfatal-warnings`, it is never enabled for Scala 3 projects and Compiler Bridge, hence we don't need to manually filter it out.

https://github.com/sbt/zinc/blob/41dd74e073efc60673cd4d161be1890e5c943685/project/HouseRulesPlugin.scala#L19-L25

https://github.com/sbt/zinc/blob/24012b9c1fdf88196841565cd1477461771140d5/build.sbt#L513-L521

For `-deprecation`, it was suppressing a bunch of deprecation warnings, hence we may not want to filter it out.

For `-YdisableFlatCpCaching`, the following logic automatically adds the flag when necessary, leading me to think that we don't need to include it in `build.sbt`

https://github.com/sbt/zinc/blob/41dd74e073efc60673cd4d161be1890e5c943685/zinc/src/main/scala/sbt/internal/inc/IncrementalCompilerImpl.scala#L443-L453

Reply:
ok. sounds good.